### PR TITLE
MKLDNN nodes factory improvements

### PR DESCRIPTION
**Pull request by vladislav-volkov**
This fix contains optimisations for MKLDNN nodes factory. Algorithm complexity for graph nodes creation reduced from O(N*N) to O(N). All changes are taken from PR 1741.

### CODE REVIEW

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_tensoriterator_node.cpp [hunk 1]
```diff
@@ -244,4 +244,4 @@ void MKLDNNTensorIteratorNode::execute(mkldnn::stream strm) {
 bool MKLDNNTensorIteratorNode::created() const {
     return getType() == TensorIterator;
 }
-REG_MKLDNN_PRIM_FOR(MKLDNNTensorIteratorNode, TensorIterator);
+REG_MKLDNN_PRIM_FOR(MKLDNNTensorIteratorNode, MKLDNNPlugin::TensorIterator);
```
Comment (dmitry-gorokhov) [inline inference-engine/src/mkldnn_plugin/nodes/mkldnn_tensoriterator_node.cpp:247]: Just wondering why do we need `MKLDNNPlugin::` here

**Comment by dmitry-gorokhov** [comment]
@ilyachur could you please take a look as well?

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_tensoriterator_node.cpp (hunk 1 above)
Comment (vladislav-volkov) [reply inference-engine/src/mkldnn_plugin/nodes/mkldnn_tensoriterator_node.cpp:247]: Without explicit namespace it clashes with TensorIterator from inference-engine/src/legacy_api/include/legacy/ie_layers.h.

Comment (ilya-lavrenov) [reply inference-engine/src/mkldnn_plugin/nodes/mkldnn_tensoriterator_node.cpp:247]: maybe it's better to remove global `using namespace InferenceEngine`?

Comment (vladislav-volkov) [reply inference-engine/src/mkldnn_plugin/nodes/mkldnn_tensoriterator_node.cpp]: removed

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**